src/main.js

```diff
@@ -54,20 +54,24 @@ window.addEventListener("DOMContentLoaded", () => {
       });
 
       // Close suggestions when clicking outside
-      document.addEventListener("click", function (event) {
-        const searchInput = document.querySelector(".search-input");
-        const suggestions = document.querySelector(".suggestions");
-
-        if (!searchInput || !suggestions) return;
-
-        const isClickInsideSearchInput = searchInput.contains(event.target);
-        const isClickInsideSuggestions = suggestions.contains(event.target);
+      document.addEventListener('click', function(event) {
```

Comment:
Could we move most of the 'search' functionality into the search.functions.js
have most of this wrapped in its own function ``searchInit()`` which bails safely if the dom element is not present. Also wrap in a try/catch for to be safe on malformed search dom structures.